### ISSUE TITLE
Can it be made compatible with steam proton/wine for linux?

### ISSUE THREAD

**Issue by MadoScientist97** [serpentiem/ddmk#301]
Not sure if this is within the project goals, but I tried using this on linux with steam proton as the newest one (7.x) added support for DMC HD collection. I thought it does not need anything extra apart from copying the files over, maybe it'll work. But ddmk is not being launched. I could try debugging it if I knew where or what to check for, but I am not that deep into proton right now. What is the most likely reason of it not being loaded? Does it absolutely depends on Directx that the vulkan translation is not helping with?

**Comment by blan44** [comment]
Also wondering about the possibility of Proton compatibility.

**Comment by MadoScientist97** [comment]
I actually somehow got it working on proton 7.0. It runs well enough for DMC3, it loads for DMC1 but crashes when fast switching weapons.

**Comment by monoai** [comment]
Hi there! I would like to contribute some information for this. I'm not sure if this should be reported in Proton or here, but I think this is the proper place for it for something I will mention later. 
**tl;dr: I think the project should use another font/include a free-to-use font.**

I got ddmk working on Proton Experimental by doing a Wine DDL Override by adding this on the launch options `WINEDLLOVERRIDES="dinput8=n,b" %command%` which could make dinput8.dll work in Proton. Launching the games however fails due to a weird font error thrown up by imgui_draw.cpp stating `Assertion failed: (0) && "Could not load font file!"`.
I fixed this by going through the files here and seeing that the project is using `consola.ttf` as its font since presumably it is one of the fonts easily included with any Windows installation, but not included within linux nor proton/wine's included fonts. You should get a copy of this font or make a link that points to a file named like this. Personally I did it by pasting a copy of consola.ttf on  `/STEAMLIBRARY/steamapps/compatdata/631510/pfx/drive_c/windows/Fonts/`

So far I got it working on DMC1, even managing to grant Dante the Yamato, and DMC3. I haven't checked too much on DMC3 since I only tested the teleporter feature, and I didn't get to check for DMC2. Assumingly it could work for these games however.
I do suggest that the project should probably include a font that it could use freely for its own uses, or use another font like `arial.ttf` or `cour.ttf` which seems to be included with Proton/Wine as default fonts.

![image](https://user-images.githubusercontent.com/46586720/191943669-7ef1b0fa-582f-44aa-a114-6ad9abf7a5b8.png)

**Comment by Dr-Qyurryus** [comment]
@monoai I tried what you suggested and it still says font error and won't load. Any Idea's?

Proton experimental, added the launch option copy pasted and downloaded consolas.ttf and pasted it in the directory you talked about.

**Comment by monoai** [comment]
@Dr-Qyurryus Hi! That is weird indeed, the only font that I could see that the project is using is `consola.ttf`so there shouldn't be any other font dependencies. Are you sure it is named that way? I see an extra **s** on your message.
Make sure you did the fix on `compatdata/631510` just in case it was overlooked, and as an alternative (and an update to the answer) you could try copying one of the symlinked fonts, paste it at the same directory, and rename it as `consola.ttf`
It should look like this:
![image](https://user-images.githubusercontent.com/46586720/198365991-9b22ff5c-1537-42cb-a2f1-62e37c962f96.png)

Hope you could report in if it works!

**Comment by monoai** [comment]
@Dr-Qyurryus Happy to hear you got it working, and even on a deck! It's nice to know it works there! It seems like `Consolas` is the proper name but the actual filename is either different or `consola.ttf` is a different font altogether. Although that topic is out of our reach and from my observation from using a symlinked font, using a different font doesn't seem to have that much of a difference in appearance. Not sure if it's because of some kind of cache or that Proton could actually imitate how a font looks like as long as you properly add a file on its place. Good work!

**Comment by yurithroddtaco** [comment]
GUYS How do I download it on the deck please teach me I’ve been trying to do it since I got the steam deck @monoai & @Dr-Qyurryus
